alt-tab-macos/ui/Application.swift

```diff
@@ -106,14 +109,41 @@ class Application: NSApplication, NSApplicationDelegate, NSWindowDelegate, NSCol
             NSApp.terminate(self)
         }
     }
+    
+    func ensureScreenRecordingCheckboxIsChecked() {
+        
+        // get first window ID
+        let window = cgWindows()[0]
+        let windowOwnerPID = window[kCGWindowNumber] // window[kCGWindowOwnerPID]
+        
+        // Try getting the image
+        let windowImage = CGWindowListCreateImage(.null, .optionIncludingWindow, windowOwnerPID as! CGWindowID, [.boundsIgnoreFraming, .bestResolution])
+        
+        // If nil, permission denied (this will also prompt for permission)
```

Comment:
If calling `CGWindowListCreateImage` prompts for permission, then why are people saying "the app doesn't work on Catalina". They should be getting the prompt if they haven't checked the checkbox, no?

Reply:
I tested with the accessiblity checkbox which I assume to work exactly the same as the screen recording checkbox. Indeed the best approach is to stop the app after prompting. On restart by the user if will either ask again, or if the user checked the box, let it proceed.